tests/helper/helper_kubectl.go

```diff
@@ -212,8 +212,10 @@ func (kubectl KubectlRunner) WaitAndCheckForTerminatingState(resourceType, names
 }
 
 func (kubectl KubectlRunner) VerifyResourceDeleted(ri ResourceInfo) {
-	session := CmdRunner(kubectl.path, "get", ri.ResourceType, "--namespace", ri.Namespace)
-	Eventually(session).Should(gexec.Exit(0))
-	output := string(session.Wait().Out.Contents())
-	Expect(output).NotTo(ContainSubstring(ri.ResourceName))
+	getOutput := func() string {
+		session := CmdRunner(kubectl.path, "get", ri.ResourceType, "--namespace", ri.Namespace)
+		Eventually(session).Should(gexec.Exit(0))
+		return string(session.Wait().Out.Contents())
+	}
+	Eventually(getOutput, 3*time.Second, 500*time.Millisecond).ShouldNot(ContainSubstring(ri.ResourceName), fmt.Sprintf("resource %s/%s should be deleted", ri.ResourceType, ri.ResourceName))
```

Comment:
This function is called to check `odo delete -w` and thus we shouldn't wait and use eventually here as the resource should be deleted by then.

Reply:
I think that, actually, the resources are not always deleted when the `odo delete -w` command returns: https://github.com/openshift/odo/issues/4733#issuecomment-847792991

And it seems not possible to make `odo` use the Kubernetes API to wait the deletion of the dependent resources, as the OpenShift RBAC policy forbids `odo` to add the `blockOwnerDeletion=true` field to the dependent resources, when ran as `developer`.

Reply:
> And it seems not possible to make odo use the Kubernetes API to wait the deletion of the dependent resources, as the OpenShift RBAC policy forbids odo to add the blockOwnerDeletion=true field to the dependent resources, when ran as developer.

OK, please leave a comment in the code for now and raise a new issue for someone to look into it later.

Reply:
I've reopened #4593  and added comments on it